pkg/discovery-manager/discovery_controller.go

```diff
@@ -47,13 +56,319 @@ type DiscoveryReconciler struct {
+							CpuStep:    flavour.Spec.Policy.Partitionable.CpuStep,
+							MemoryStep: flavour.Spec.Policy.Partitionable.MemoryStep,
+						}
+					}
+					return nil
+				}(),
+				Aggregatable: func() *nodecorev1alpha1.Aggregatable {
+					if flavour.Spec.Policy.Aggregatable != nil {
+						return &nodecorev1alpha1.Aggregatable{
+							MinCount: flavour.Spec.Policy.Aggregatable.MinCount,
+							MaxCount: flavour.Spec.Policy.Aggregatable.MaxCount,
+						}
+					}
+					return nil
+				}(),
+			},
+			Owner: nodecorev1alpha1.NodeIdentity{
+				Domain: flavour.Spec.Owner.Domain,
+				IP:     flavour.Spec.Owner.IP,
+				NodeID: flavour.Spec.Owner.NodeID,
+			},
+			Price: nodecorev1alpha1.Price{
+				Amount:   flavour.Spec.Price.Amount,
+				Currency: flavour.Spec.Price.Currency,
+				Period:   flavour.Spec.Price.Period,
+			},
+			OptionalFields: nodecorev1alpha1.OptionalFields{
+				Availability: flavour.Spec.OptionalFields.Availability || false,
+			},
+		},
+		Status: nodecorev1alpha1.FlavourStatus{
+			LastUpdateTime: time.Now().String(),
+		},
+	})
+
+	return flavourCR
+}
+
+// createPeeringCandidateCustomResources creates a PeeringCandidate CR from a Flavour and a Discovery
+func createPeeringCandidateCustomResources(flavourPeeringCandidate *nodecorev1alpha1.Flavour, discovery *advertisementv1alpha1.Discovery, seq int) *advertisementv1alpha1.PeeringCandidate {
+	return (&advertisementv1alpha1.PeeringCandidate{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      forgePeeringCandidateName(flavourPeeringCandidate.Name, seq),
+			Namespace: "default",
+		},
+		Spec: advertisementv1alpha1.PeeringCandidateSpec{
+			SolverID: discovery.Spec.SolverID,
+			Flavour:  *flavourPeeringCandidate,
+		},
+	})
+}
+
+// createReservationCustomResource creates a Reservation CR from a PeeringCandidate
+func createReservationCustomResource(peeringCandidate advertisementv1alpha1.PeeringCandidate) *reservationv1alpha1.Reservation {
+	reservationCR := (&reservationv1alpha1.Reservation{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "reservation-" + peeringCandidate.Spec.Flavour.Name,
+			Namespace: "default",
+		},
+		Spec: reservationv1alpha1.ReservationSpec{
+			Buyer: nodecorev1alpha1.NodeIdentity{
+				NodeID: CLIENT_ID,
+			},
+			Seller: nodecorev1alpha1.NodeIdentity{
+				NodeID: peeringCandidate.Spec.Flavour.Spec.Owner.NodeID,
+			},
+			FlavourID: peeringCandidate.Spec.Flavour.Name,
+			PeeringCandidate: nodecorev1alpha1.GenericRef{
+				Name:      peeringCandidate.Name,
+				Namespace: peeringCandidate.Namespace,
+			},
+			Reserve:  true,
+			Purchase: true,
+		},
+	})
+
+	return reservationCR
+}
+
+// parseSelectorValues parses the values of the selector in int64 (REAR only supports int64 for now)
+func parseSelectorValues(selector nodecorev1alpha1.FlavourSelector) (cpuInt, ramInt, ephStorage int64) {
+	cpuInt, _ = selector.Cpu.AsInt64()
+	ramInt, _ = selector.Memory.AsInt64()
+	ephStorage, _ = selector.EphemeralStorage.AsInt64()
+	return
+}
```

Comment:
you can not return back directly the map?

Reply:
Fixed, also moved in `utils/parseutil`